src/library.js

```diff
@@ -402,7 +402,7 @@ addToLibrary({
   // so we cannot override parts of it, and therefore cannot use libc_optz.
 #if (SHRINK_LEVEL < 2 || LINKABLE || process.env.EMCC_FORCE_STDLIBS) && !STANDALONE_WASM && !BULK_MEMORY
 
-#if MIN_CHROME_VERSION < 45 || MIN_EDGE_VERSION < 14 || MIN_FIREFOX_VERSION < 34 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED || MIN_SAFARI_VERSION < 100101
+#if caniuse('js.TypedArray.copyWithin')
```

Comment:
Shouldn't this be flipped? The old checks were "if we must support chrome before 45 etc. then add polyfill" but now it reads as "if I can use the feature then add polyfill".

Reply:
Yup, good catch. I made similar mistake in other places before.

```suggestion
#if !caniuse('js.TypedArray.copyWithin')
```

Reply:
Sounds good. I can do a careful read to look for similar mistakes - is now a good time, or should I wait until it is no longer a draft?

Reply:
I guess let me get around to addressing existing issues first, and then I'll re-request review. Since this changes a lot of files, extra eyes will certainly be welcome.